realm/realm-library/src/androidTestObjectServer/java/io/realm/SchemaTests.java

```diff
@@ -145,4 +147,54 @@ public void allow_addField() {
 
         realm.close();
     }
+
+    @Test
+    public void addPrimaryKey_notAllowed() {
+        String className = "StringOnly";
+        Realm realm = Realm.getInstance(config);
+
+        realm.beginTransaction();
+        RealmObjectSchema objectSchema = realm.getSchema().get(className);
+        objectSchema.addField("foo", String.class);
+
+        try {
+            objectSchema.addPrimaryKey("foo");
```

Comment:
Because migration of primary key in non-sync Realm would be quite a pain if you can't remove and re-add primary key to a different field. Mapping all objects to a new object schema just to change PK is rough...